catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/UiConfigurationPropertiesFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.ui.searchui.simple.properties;
+
+import java.io.IOException;
+import java.util.Dictionary;
+import java.util.Optional;
+import org.codice.ddf.branding.BrandingRegistry;
+import org.codice.ddf.configuration.DictionaryMap;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/* Looks up the configuration for the UI and creates an object to represent it*/
+public class UiConfigurationPropertiesFactory {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(UiConfigurationPropertiesFactory.class);
+
+  private static final String CONFIG_PID = "ddf.platform.ui.config";
+
+  private static UiConfigurationPropertiesFactory uniqueInstance =
+      new UiConfigurationPropertiesFactory();
+
+  private Optional<BrandingRegistry> branding = Optional.empty();
+
+  private ConfigurationAdmin configurationAdmin;
+
+  private UiConfigurationPropertiesFactory() {}
+
+  /**
+   * @return a new UiConfigurationProperties object that contains the properties to build out this
+   *     UI
+   */
+  public UiConfigurationProperties getProperties() {
+    Dictionary<String, Object> properties;
+    try {
+      if (configurationAdmin.listConfigurations(String.format("(service.pid=%s)", CONFIG_PID))
+          != null) {
+        properties = configurationAdmin.getConfiguration(CONFIG_PID, null).getProperties();
```

Comment:
✏️  We have seen weird issues manifest when using this `getConfiguration` method because if it doesn't find a configuration it creates an empty one for you. I'd recommend storing the above `listConfigurations` call to a variable and store it, then reuse it here. I don't like that you have to store the array of `Configuration`s but I think it' might be better than getting it and throwing it away.

Reply:
If I'm not mistaken, passing the location as `null` should prevent that issue from happening.

Reply:
That's true. I don't think there's any functional issue with this code. I just get really defensive against Config Admin because we've been burned by it so many times. I think whether or not you use the original `listConfigurations` call is semantical. This is nitpicky, so I'm going to approve and leave it up to you 😄

Reply:
For what it's worth, you being cautious about it is absolutely valid here as when I was testing this I noticed that I was creating that config in my bundle's context when I didn't have that null there.